src/ggml-cuda.cu

```diff
@@ -8233,6 +8416,12 @@ bool ggml_cuda_compute_forward(struct ggml_compute_params * params, struct ggml_
             }
             func = ggml_cuda_mul_mat;
             break;
+        case GGML_OP_MUL_MAT_ID:
+            if (!any_on_device && !ggml_cuda_can_mul_mat(tensor->src[0], tensor->src[1], tensor)) {
```

Comment:
Isn't `tensor->src[0]` the tensor with the indices?

Reply:
Yes, it's a bug, already fixed locally.